mpas_xarray.py

```diff
@@ -54,6 +54,34 @@ def preprocess_mpas(ds, yearoffset=1700): #{{{
 
     return ds #}}}
 
+def preprocess_mpas_timeSeriesStats(ds, yearoffset=1849): #{{{
+    """
+    Builds corret time specification for MPAS time series stats am fields,
+    allowing a year offset because the time must be betwee 1678 and 2262 
+    based on the xarray library.
+
+    Milena Veneziani
+    04/18/2016
+    """
+
+    daysSinceStart = ds.timeSeriesStatsMonthly_avg_daysSinceStartOfSim_1
+    time = [datetime.datetime(yearoffset,12,31) + datetime.timedelta(x) for x in daysSinceStart.values]
+    datetimes = pd.to_datetime(time)
+    # make sure  date times are set up properly
+    assert datetimes[0].year > 1678, 'ERROR: yearoffset=%s'%(yearoffset) + \
+            ' must be large enough to ensure datetimes larger than year 1678'
+    assert datetimes[-1].year < 2262, 'ERROR: yearoffset=%s'%(yearoffset) + \
+            ' must be large enough to ensure datetimes larger than year 2262'
```

Comment:
This code is duplicated with the code above in the function.  Probably this set of asserts should be written as a function for cleaner code.